src/test/java/com/qiyana/CustomerDaoTest.java

```diff
@@ -0,0 +1,101 @@
+package com.qiyana;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.when;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.EntityTransaction;
+import javax.persistence.Persistence;
+import org.junit.jupiter.api.*;
+import org.qiyana.DAO.imp.CustomerDao;
+import org.qiyana.model.Customer;
+
+public class CustomerDaoTest {
+  static EntityManagerFactory etf;
+  static EntityManagerFactory etfMock;
+  static EntityManager emMock;
+  static EntityTransaction transaction;
+  static CustomerDao customerDao;
+  static Customer customer;
+  static CustomerDao customerDaoMock;
+
+  @BeforeAll
+  public static void init() {
+    etf = Persistence.createEntityManagerFactory("persistence");
+    customerDao = new CustomerDao(etf);
+    etfMock = mock(EntityManagerFactory.class);
+    transaction = mock(EntityTransaction.class);
+    emMock = mock(EntityManager.class);
+    when(etfMock.createEntityManager()).thenReturn(emMock);
+    when(emMock.getTransaction()).thenReturn(transaction);
+    customerDaoMock = new CustomerDao(etfMock);
+  }
+
+  @BeforeEach
+  public void setUp() {
+    customer =
+        new Customer(null, "QiyanaTech", "test@gmail.com", "Rua Tiberius Dourado", "123456789");
+    customerDao.add(customer);
+  }
+
+  @Test
+  void addTest_whenCustomerCorrectlyAdded_mustNotThrowException() {
+    assertNotNull(customer.getId());
+  }
+
+  @Test
+  void addTest_shouldThrowException() {
+    doThrow(new IllegalArgumentException()).when(emMock).persist(any());
+    assertThrows(IllegalArgumentException.class, () -> customerDaoMock.add(customer));
+  }
+
+  @Test
+  void getTest_shouldReturnCustomer() {
+    assertNotNull(customerDao.get(customer.getId()));
+  }
+
+  @Test
+  void getTest_shouldThrowException() {
+    doThrow(new IllegalArgumentException()).when(emMock).find(any(), any());
+    assertThrows(IllegalArgumentException.class, () -> customerDaoMock.get(customer.getId()));
+  }
+
+  @Test
+  void getAllTest_shouldReturnAllCustomer() {
+    assertNotNull(customerDao.getAll());
+  }
+
+  @Test
+  void getAllTest_shouldThrowException() {
+    when(emMock.getCriteriaBuilder()).thenThrow(new IllegalArgumentException());
+    assertThrows(IllegalArgumentException.class, customerDaoMock::getAll);
+  }
+
+  @Test
+  void updateTest_whenCustomerCorrectlyUpdated_mustNotThrowException() {
+    Customer customerUpdated =
+        new Customer(
+            customer.getId(), "QiyanaTech1", "teste789@gmail.com", "Rua Ixtal", "1254785336");
+    assertDoesNotThrow(() -> customerDao.update(customerUpdated));
```

Comment:
While this assertion works for testing the exception behavior, it would be a better test of the `update` operation if, after updating, you attempt to `get` using `customer.getId()`, then confirm/assert that whatever you updated is also updated in the `get` result. Does that make sense?

(e.g. if you updated the customer's name, try to `get` by ID and assert that the get result's name equals the expected new customer name)

Reply:
The same is valid for other `update` entity tests.